library/foreman_provisioner.py

```diff
@@ -0,0 +1,292 @@
+   password:
+     description:
+         - Password of login user
+     required: true
+   auth_url:
+     description:
+         - The Foreman api url
+     required: true
+   host_id:
+     description:
+         - Name or ID of the host as listed in foreman
+     required: true
+   rebuild:
+     description:
+         - Should we rebuilt the requested host
+     default: true
+     required: false
+   mgmt_strategy:
+     description:
+         - Whether to use Foreman or system ipmi command.
+     default: 'foreman'
+     required: false
+   mgmt_action:
+     description:
+         - Which command to send with the power-management selected by
+         mgmt_strategy. For example: reset, reboot, cycle
+     default: 'cycle'
+     required: false
+   wait_for_host:
+     description:
+         - Whether we should wait for the host given the 'rebuild' state was set.
+     default: true
+     required: false
+'''
+
+
+MIN_SUPPORTED_VERSION = 2
+MGMT_SUPPORTED_STRATEGIES = ['foreman', 'ipmi']
+WAIT_TO_FINISH_BUILDING = 10
+WAIT_TO_FINISH_BOOTING = 10
+
+
+class ForemanManager(object):
+    """
+    This class represents a simple interface for foreman* to easily rebuild /
+    get / reserve hosts from foreman.
+    *Foreman: http://theforeman.org/
+    """
+    def __init__(self, url, username, password, extra_headers=None, version=2):
+        """
+        :param url: the url of the foreman we wish to authenticate with
+        :param username: the username we will use to login
+        :param password: the password we will use to login
+        :param extra_headers: if we require extra headers to be added to the
+        http request
+        :param version: the version of foreman API we wish to use (default: 2)
+        :type version: int
+        """
+        if version < MIN_SUPPORTED_VERSION:
+            raise Exception("API version: {0} "
+                            "is not supported at the moment".format(version))
+
+        self.session = requests.Session()
+        self.session.auth = (username, password)
+
+        headers = {'Accept': 'application/json',
+                   'Content-type': 'application/json'}
+
+        if extra_headers:
+            headers.update(extra_headers)
+
+        self.session.headers.update(headers)
+
+        self.url = url.rstrip('/')
+        self.default_uri = '/api/v2/hosts/'
+
+    def reserve_host(self, host_id):
+        """
+        This method 'tags' a host as reserved in foreman
+        :param host_id: the name of ID of the host we wish to reserve
+        :returns: the host information on success, else empty body
+        :rtype: list of dictionaries -- [{"host": {}}]
+        """
+        #TODO(tkammer): add the option to provide the query itself after "?"
+        request_url = '{0}/api/hosts_reserve' \
+                      '?query=name ~ {1}'.format(self.url, host_id)
+        response = self.session.get(request_url, verify=False)
+        body = response.json()
+        return body
```

Comment:
Why do we need the returned body?
If you don't use it, please check the returned code from the response and raise an exception in case of failure.
The methods don't have to return value, especially it there's is no use in it...